File: scratch/parse_test_failures.php
<?php

$logPath = 'C:\\Users\\Acer\\.gemini\\antigravity-ide\\brain\\4d4dc69e-c458-4814-a874-352a616e23ba\\.system_generated\\tasks\\task-889.log';
$content = file_get_contents($logPath);

preg_match_all('/FAILED\s+(Tests\\\\[^\s]+)\s+>\s+([^\n]+)/', $content, $matches, PREG_SET_ORDER);

echo "Total Failed Tests Found: " . count($matches) . "\n\n";

$failures = [];
foreach ($matches as $match) {
    $failures[] = [
        'class' => trim($match[1]),
        'method' => trim($match[2]),
    ];
}

file_put_contents('scratch/failure_list.json', json_encode($failures, JSON_PRETTY_PRINT));
echo json_encode($failures, JSON_PRETTY_PRINT);

File: scratch/parse_test_failures.py
import re
import json

log_path = r"C:\Users\Acer\.gemini\antigravity-ide\brain\4d4dc69e-c458-4814-a874-352a616e23ba\.system_generated\tasks\task-793.log"

with open(log_path, 'r', encoding='utf-8', errors='ignore') as f:
    content = f.read()

# Find all FAILED lines
# Example line in log:
# FAILED  Tests\Feature\Module6MaintenanceTest > test_prior_status_restored_when_maintenance_deleted
pattern = r"FAILED\s+(Tests\\[^\s]+)\s+>\s+([^\n]+)"
matches = re.findall(pattern, content)

print(f"Total Failed Tests Found: {len(matches)}")
failures = []
for test_class, test_method in matches:
    failures.append({
        'class': test_class.strip(),
        'method': test_method.strip()
    })

print(json.dumps(failures, indent=2))
